control-plane/agents/src/bin/core/controller/scheduling/volume_policy/pool.rs

```diff
@@ -56,7 +56,11 @@ impl PoolBaseFilters {
     }
     /// Should only attempt to use encrypted pools.
     pub(crate) fn encrypted(request: &GetSuitablePoolsContext, item: &PoolItem) -> bool {
-        request.encrypted() == item.pool.encrypted
+        let use_this = request.encrypted() == item.pool.encrypted;
+        if !use_this && request.registry().encryption_preference_soft() {
+            return !use_this;
+        }
+        use_this
```

Comment:
We should do this only for encrypted volumes?

```suggestion
        let use_this = request.encrypted() == item.pool.encrypted;
        if !use_this && request.encrypted() && request.registry().encryption_preference_soft() {
            return true;
        }
        use_this
```

Reply:
> We should do this only for encrypted volumes?

The current logic will give benefit to non-encypted volumes too, in case non-encrypted pools are scarce. e.g stsAffinityGroup I was testing scales up fine from 1 -> 2 replica, which otherwise will fail. wdyt.

Reply:
possibly could be useful yes though may not be always desireable and we'd need to change the sorting to prefer the correct one, ie encrypted pools to encrypted volumes, with soft fallback to non-encrypted pools and non-encrypted pools to non-encrypted volumes with fallback to encrypted pools

And if so, maybe this should then become a multiple option, rather than a single bool

Reply:
ok, let's not complicate with multiple option. Will restrict this to only encrypted.